### DISCUSSION TITLE
[solved] How to remove old branches?

### DISCUSSION THREAD

**Discussion by humdingerb** [Genio-The-Haiku-IDE/Genio#423] [Q&A]
Over time, I have quite a lot old remote branches still showing under "Source control". All have long been deleted locally and on the remote GitHub. How do I get rid of them in Genio?

ETA: the "Fetch prune" etc. commands in Genio don't yet works AFAIK, letting me enter credentials but failung with, "username does not match previous request". Maybe because of missing SSL authentication?

Anyway, still, is there a way to remove those sale branches?

**Comment by lonemadmax** [comment]
> letting me enter credentials but failung with, "username does not match previous request". Maybe because of missing SSL authentication?

It seems for ssh the password is checked to be filled but nothing is done with it. So, even if an ssh-agent is supposed to be called from within libgit2, something is fishy there.

I guess "run `git fetch --prune` from the terminal and Genio will automatically update its list" is not the answer you are looking for. I use different push and fetch URL for the remote so that fetching does not need authentication. You can change them through `git remote set-url [--push] remote_name url`.
```
$ git remote -v
origin	https://github.com/Genio-The-Haiku-IDE/Genio.git (fetch)
origin	git@github.com:Genio-The-Haiku-IDE/Genio.git (push)
```

**Comment by humdingerb** [comment]
Doing a `git fetch --prune` from Terminal doesn't change the remote branches shown in Genio. TBH, I never did a `git fetch --prune` before, just doing `git branch -D` when a PR is merged, or trashing old branches in the browser at the GitHub site...

Maybe I haven't made myself clear before: all these remote branches shown in Genio do not exist any more at GitHub.

ETA: using the HTTPS url for fetching doesn't work either. Same alert after typing in login and password.

**Reply by lonemadmax** [reply]
> Doing a `git fetch --prune` from Terminal doesn't change the remote branches shown in Genio. TBH, I never did [...]

TBH too, I have to look it up whenever I want to do it. Did you include the name of the remote in the command? Genio shows the branches and tags that are in the (local copy of the) repo, so if you prune them, it won't see them.

You can use `git branch -D` with remote tracking branches too, if you add the `-r` option.

> using the HTTPS url for fetching doesn't work either. Same alert after typing in login and password.

Works for me. Genio only fetches from 'origin' remote, may that be the problem?

> Maybe I haven't made myself clear before: all these remote branches shown in Genio do not exist any more at GitHub.

You did. You see remote branches that the remote doesn't have anymore. That's what prune removes.

Feel free to include (or send me privately through mail) the output of `git remote -v` and `git branch -a`.

**Comment by humdingerb** [comment]
> Did you include the name of the remote in the command?

Aha! That was it, adding the remote name resulted in actual pruning!
Thanks a lot!
